### ISSUE TITLE
NodeRed 3.0: Subnodes: Number fields do not accept subnode env variables anymore

### ISSUE THREAD

**Issue by mbbrenner** [node-red/node-red#3795]
### Current Behavior

After update to NodeRed 3, following problem occurs:

NOK: Fields of type number produce error "Invalid property expression: ${XYZ}"
OK: No error message for fields of type string....

Anyway beside warnings during deploy, it seems to work... So it might be just some kind of checking...

### Expected Behavior

Accept env variables also in number fields w/o error

### Steps To Reproduce

- have a subflow
- define env variable XYZ
- use a inject node
- set field type to number
- enter ${XYZ} as value

### Example flow

```
[{"id":"1e7b764a.a13622","type":"inject","z":"1ee255b8.1dd56a","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"},{"p":"id","v":"${BLINDS_NSP_ID}","vt":"num"},{"p":"name","v":"${BLINDS_NAME}","vt":"str"}],"repeat":"","crontab":"","once":true,"onceDelay":"1","topic":"Init","payload":"","payloadType":"date","x":140,"y":300,"wires":[["abd425b1.0fb3b"]]}]
```

### Environment

- Node-RED version: 3
- Node.js version:16
- npm version:
- Platform/OS: x64 official docker image
- Browser: Safari 15.5

**Comment by knolleary** [comment]
We introduced better error reporting in various places in 3.0 when there are invalid values in the typedInputs. It didn't take into account the fact an Env Var string (`${FOO}`) should be allowed.

**Comment by mbbrenner** [comment]
Hi NodeRed Team,

are there any news on this? I did not find it in  any latest release. Would be cool if that could be fixed soon.

Thanks a lot

Matthias

**Comment by knolleary** [comment]
Fixed via #4582 - will be in Node-RED 3.1.6
